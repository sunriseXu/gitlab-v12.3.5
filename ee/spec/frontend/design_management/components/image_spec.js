import { shallowMount } from '@vue/test-utils';
import DesignImage from 'ee/design_management/components/image.vue';

describe('Design management large image component', () => {
  let wrapper;

  function createComponent(propsData) {
    wrapper = shallowMount(DesignImage, {
      sync: false,
      propsData,
    });
  }

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders loading state', () => {
    createComponent({
      isLoading: true,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders image', () => {
    createComponent({
      isLoading: false,
      image: 'test.jpg',
      name: 'test',
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
