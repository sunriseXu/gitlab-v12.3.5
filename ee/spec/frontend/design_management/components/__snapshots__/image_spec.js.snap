// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design management large image component renders image 1`] = `
<div
  class="d-flex align-items-center h-100 w-100 p-3 overflow-hidden js-design-image"
>
  <img
    alt="test"
    class="ml-auto mr-auto img-fluid mh-100 design-image"
    src="test.jpg"
  />
</div>
`;

exports[`Design management large image component renders loading state 1`] = `
<div
  class="d-flex align-items-center h-100 w-100 p-3 overflow-hidden js-design-image"
  isloading="true"
>
  <img
    alt=""
    class="ml-auto mr-auto img-fluid mh-100 design-image"
    src=""
  />
</div>
`;
